src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+
+1. git clone https://github.com/stella-z3g/RubricEval -b public_release
+2. cd RubricEval/ && pip install -e .
+3. python scripts/convert_dataset_to_rubriceval_input_format.py  # This will generate an example instructions.json
+4. rubric_eval get_rubrics --input_path=instructions.json --output_path=instructions_with_rubrics.json
+5. rubric_eval get_completions --model_configs="['gpt-4o-2024-05-13']" --input_path=instructions_with_rubrics.json --output_path_prefix=output_
+6. rubric_eval --input_path_prefix=output_ --output_path_prefix=eval_output_
+"""
+
+
+import sys
+import os
+import ast
+from pathlib import Path
+import pandas as pd
+from typing import List
+from alpaca_eval import utils as ae_utils
+
+import fire
+
+from .helper import get_detailed_rubrics, get_model_completions, get_evaluations
+from .annotators import RubricBrainstormer
+
+CUR_DIR = Path(__file__).parent
+
+__all__ = ["get_rubrics", "get_outputs", "eval_outputs"]
+
+
+def check_df_fields(df, required_fields, optional_fields=set()):
+    allowed_fields = required_fields | optional_fields
+    
+    actual_fields = set(df.columns)
+    
+    # Check if all required fields are present
+    if not required_fields.issubset(actual_fields):
+        missing = required_fields - actual_fields
+        raise RuntimeError(f"Missing required fields: {missing}")
+    
+    # Check if there are any fields not in the allowed set
+    extra_fields = actual_fields - allowed_fields
+    if extra_fields:
+        raise RuntimeError(f"Extra fields found: {extra_fields}")
+    
+
+def cleanup_df_rubrics(df_rubrics):
+    # Drop rows where the "scoring_scales" column is not a dict (because sometimes rubric generator will refuse to generate rubric)
+    mask = df_rubrics["scoring_scales"].apply(lambda x: isinstance(x, dict))
+    return df_rubrics[mask]
+
+
+def list_files_with_prefix(path_prefix, suffix):
+    matching_files = []
+    
+    # Expand path_prefix to full absolute path
+    full_path_prefix = os.path.abspath(path_prefix)
+    
+    # Get the directory part of the full_path_prefix
+    prefix_dir = os.path.abspath(os.path.dirname(full_path_prefix))
+    
+    # Walk through the directory
+    for root, _, files in os.walk(prefix_dir):
+        for file in files:
+            full_path = os.path.join(root, file)
+            if (
+                full_path == os.path.join(prefix_dir, file)
+                and full_path.startswith(full_path_prefix)
+                and full_path.endswith(suffix)
+            ):
+                matching_files.append(full_path)
+    
+    return matching_files
+
+
+def string_to_list(s):
+    try:
+        return ast.literal_eval(s)
+    except (ValueError, SyntaxError):
+        # Handle the case where the string is not a valid literal
+        print(f"Error: '{s}' is not a valid list representation")
+        return None
+
+
+
+def get_rubrics(
+    input_path: str = "instructions.json",
+    output_path: str = "instructions_with_rubrics.json",
```

Comment:
check main.py in alpaca_eval: the types should be `Union[AnyPath, str]` instead of just str. 

I would also suggest having a default output_path (e.g. if it's `None`) that will use the same as the input path but with a prefix / suffix such as `*_with_rubrics`. I have no strong preference for prefix vs suffix but I think we should be consistent for all the steps in the pipeline (completion / reubrics / eval ...)


Baiscally by default `rubric_eval get_rubrics --input_path=instructions.json` should save to an output path that makes sense => decrease the amount of work for the user

Reply:
Just checked and it seems `AnyPath` already includes `str` haha (https://github.com/tatsu-lab/alpaca_eval/blob/0b36023e90b3fa28b9fc785eac5de43f3288ac7d/src/alpaca_eval/types.py#L9), maybe here we just use `AnyPath`?

Reply:
> I think we should be consistent for all the steps in the pipeline (completion / reubrics / eval ...)

Agreed - I updated the PR to try to make it consistent. The updated design is that:
- `get_rubrics` by default takes `--input_path=instructions.json` and outputs `instructions_with_rubrics.json`
- `get_completions` by default takes `--input_path=instructions_with_rubrics.json` and outputs `completions.json`
- `evaluations` by default takes `--input_path=completions.json` and outputs `evaluations.json`
Is this an okay design?